### PR TITLE
fix: puzzle transition

### PR DESCRIPTION
**Pull request by enkot**
### 🔗 Linked issue

fix https://github.com/unjs/website/issues/37

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme, or JSdoc annotations)
- [x] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [x] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- Simplified https://github.com/unjs/website/blob/main/components/puzzle/Puzzle.vue, removed magic numbers.
- Use `pointer-events-none` on child div to remove "doom flickers" (https://www.joshwcomeau.com/animation/css-transitions/#doom-flicker-13)
- Use transforms for better performance.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: components/puzzle/Puzzle.vue [hunk 1]
```diff
@@ -6,31 +6,28 @@ const props = defineProps<{
...
-      <div class="puzzle-row">
-        <PuzzlePart v-for="(part, index) in firstLine" :key="part.name" :part="part" :class="{ 'relative left--6 rotate--20 group-hover:left-0 group-hover:rotate-0': index === 0 }" />
-      </div>
-      <div class="puzzle-row">
-        <PuzzlePart v-for="part in secondLine" :key="part.name" :part="part" />
-      </div>
-      <div class="puzzle-row">
-        <PuzzlePart v-for="part in thirdLine" :key="part.name" :part="part" />
+  <div relative flex justify-end w-84>
+    <div flex="~ col" space-y--9 :class="triggered ? 'drop-shadow-xl' : 'drop-shadow-md'">
```
Comment (Barbapapazes) [inline components/puzzle/Puzzle.vue:26]: I think we need a transition on this.

**Review by Barbapapazes — CHANGES_REQUESTED** [review]

**Comment by Barbapapazes** [comment]
It's very cleaver to use a fixed square for handling the hover! And love the refactor 💛

File: components/puzzle/Puzzle.vue [hunk 2]
```diff
@@ -6,31 +6,28 @@ const props = defineProps<{
...
+      <div v-for="(row, i) in rows" :key="i" flex flex-row space-x--9 transition-transform ease-in duration-200>
+        <PuzzlePart v-for="(part, j) in row" :key="part.name" :part="part" :class="{ [transforms[i][j]]: !triggered, '<lg:opacity-0': !j }" shrink-0 />
       </div>
     </div>
-    <!-- 27px and 23px are magics numbers -->
-    <PuzzlePart opacity-0 lg:opacity-100 absolute bottom--20 xl:bottom--12 rotate--25 class="left-[calc(2*-136px)] xl:left-[calc(4*-136px)] peer-hover:bottom-[calc(1*136px-27px)] peer-hover:left-[calc(0.5*-136px-23px)]" peer-hover:rotate-0 :part="away" />
+    <div absolute inset-0 @mouseenter="triggered = true" @mouseleave="triggered = false" />
```
Comment (Barbapapazes) [inline components/puzzle/Puzzle.vue:31]: Couldn't we have a `peer` selector (moving the absolute before the puzzle ? https://tailwindcss.com/docs/hover-focus-and-other-states#styling-based-on-sibling-state

**Comment by Barbapapazes** [comment]
On lg screen, you need to move the away part a little to the right!

image

**Comment by enkot** [comment]
@Barbapapazes Made it work with `group-hover`, `pointer-events-none` on child div did the trick :)

**Comment by Barbapapazes** [comment]
> @Barbapapazes Made it work with `group-hover`, `pointer-events-none` on child div did the trick :)

oh niiice!

**Review by Barbapapazes — APPROVED** [review]
Looks perfect! Thank you so much for this!

**Pull request merged**